### ISSUE TITLE
zIndex not working on new architecture enabled

### ISSUE THREAD

**Issue by grgmo** [facebook/react-native#38513]
### Description

`zIndex` does not seem to work on `ios` or `android` with new architecture enabled.

It works as expected for old architecture.

I just have 2 `View`s with `Text` inside it. `View` on top has `zIndex: 1` and bottom one does not have `zIndex` set. 

This bug cannot be reproduced using `react-native` in browser snack.

The bug was previously raised here https://github.com/facebook/react-native/issues/34617 but seems to have been closed.

**Expected** 
![Simulator Screenshot - iPhone 14 Pro - 2023-07-19 at 11 04 12](https://github.com/facebook/react-native/assets/6507800/a84dddae-0515-4bde-b90e-5c0450f8986f)

**Actual**
![Simulator Screenshot - iPhone 14 Pro - 2023-07-19 at 10 56 03](https://github.com/facebook/react-native/assets/6507800/510c111c-0582-40bf-832b-e15d6f7f0eed)

### React Native Version

0.72.3

### Output of `npx react-native info`

System:
  OS: macOS 13.3.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 96.30 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 16.14.2
    path: ~/.nvm/versions/node/v16.14.2/bin/node
  Yarn:
    version: 1.22.19
    path: ~/.nvm/versions/node/v16.14.2/bin/yarn
  npm:
    version: 8.5.0
    path: ~/.nvm/versions/node/v16.14.2/bin/npm
  Watchman:
    version: 2023.02.06.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /Users/gurung/.asdf/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK:
    API Levels:
      - "30"
      - "31"
      - "32"
      - "33"
    Build Tools:
      - 30.0.3
      - 31.0.0
      - 32.0.0
      - 33.0.0
    System Images:
      - android-30 | Google APIs ARM 64 v8a
      - android-31 | Google APIs ARM 64 v8a
      - android-31 | Google Play ARM 64 v8a
      - android-32 | Google APIs ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2021.1 AI-211.7628.21.2111.8309675
  Xcode:
    version: 14.3/14E222b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.17
    path: /usr/bin/javac
  Ruby:
    version: 3.0.3
    path: /Users/gurung/.asdf/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.3
    wanted: 0.72.3
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true

### Steps to reproduce

create a new project

`npx react-native@latest init AwesomeProject`

Follow steps to enable new architecture https://reactnative.dev/docs/next/new-architecture-app-intro

in `App.tsx` anywhere

```
      <View
        style={{
          height: 20,
          backgroundColor: 'red',
          marginBottom: -10,
          zIndex: 1,
        }}>
        <Text>I SHOULD BE ON TOP</Text>
      </View>
      <View
        style={{
          height: 20,
          backgroundColor: 'pink',
        }}>
        <Text>I SHOULD BE BEHIND</Text>
      </View>
```

**Expected**
`I SHOULD BE ON TOP` text should be displayed on top of `I SHOULD BE BEHIND`

**Actual**
`I SHOULD BE BEHIND` is displayed on top of `I SHOULD BE ON TOP`

### Snack, code example, screenshot, or link to a repository

https://github.com/grgmo/react-native-zindex-repro

**Comment by grgmo** [comment]
> ⚠️	Missing Reproducible Example
> ℹ️	We could not detect a reproducible example in your issue report. Please provide either:
> * If your bug is UI related: a [Snack](https://snack.expo.dev)
> * If your bug is build/update related: use our [Reproducer Template](https://github.com/react-native-community/reproducer-react-native/generate)

This bug is not reproducible on Snack. Steps to reproduce and code snippet is mentioned above

**Comment by cortinico** [comment]
> This bug is not reproducible on Snack. Steps to reproduce and code snippet is mentioned above

Then please provide a reproducer project using the template as suggested

**Comment by grgmo** [comment]
> > This bug is not reproducible on Snack. Steps to reproduce and code snippet is mentioned above
> 
> Then please provide a reproducer project using the template as suggested

my bad, attached link to the repo

**Comment by meghvarse** [comment]
Facing same issue, zIndex not working after enable new architecture.

**Comment by meghvarse** [comment]
@cortinico , Can you please provide any update on this issue?

**Comment by szmarczak** [comment]
Same here.

**Comment by szmarczak** [comment]
> Then please provide a reproducer project using the template as suggested

The guy literally has provided code you can copy-paste and you didn't even bother to take half a minute to bootstrap the project yourself. Let's not be crazy, those requirements are counter-productive. Snack is not an environment to test bugs, and bootstrapping a rn project on your own shouldn't be that hard.

**Comment by szmarczak** [comment]
Looks like this issue is a duplicate of #33866

**Comment by javache** [comment]
This has been resolved in https://github.com/facebook/react-native/commit/850349b1d274f0cc2595eee2f6bb361a958bb2e2, which will go out with v73.

**Comment by vincentmegia** [comment]
i tried to applied the patch locally, not working for me.

**Comment by joemun** [comment]
> What about android?

Should we reopen this issue? or create a new issue for z-index not working in Android? @cortinico

**Comment by cortinico** [comment]
As @javache mentioned, this fix is shipped in 0.73. Asking about Android makes no sense here. Let's wait for the 0.73 stable and open a new issue if the problem persists on 0.73.0

**Comment by joevilches** [comment]
A potential workaround for folks that have this problem would be to add `position: relative` to the style of the View with the `zIndex` applied. Technically this will have some other potential changes since this will make it so that we are now forming a stacking context so a child could change its stacking order a bit, but I assume that is fairly rare.
